### PR TITLE
WIP: Run `kubeadm init phase addon all` to ensure kube-proxy is up in v1.13+

### LINKED ISSUE #3936: waiting for k8s-app=kube-proxy: timed out waiting for the condition
minikube start
😄  minikube v0.35.0 on darwin (amd64)
💡  Tip: Use 'minikube start -p <name>' to create a new cluster, or 'minikube delete' to delete this one.
🔄  Restarting existing virtualbox VM for "minikube" ...
⌛  Waiting for SSH access ...
📶  "minikube" IP address is 192.168.99.100
🐳  Configuring Docker as the container runtime ...
✨  Preparing Kubernetes environment ...
🚜  Pulling images required by Kubernetes v1.13.4 ...
🔄  Relaunching Kubernetes v1.13.4 using kubeadm ... 
⌛  Waiting for pods: apiserver proxy
💣  Error restarting cluster: wait: waiting for k8s-app=kube-proxy: timed out waiting for the condition

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new

### PR DESCRIPTION
**Pull request by liubog2008**
fix #3936 #3843

I find that kube-proxy is not created after `minikube start`(restart) so that `waitForPods` always waits until timeout.

`kubeadm init phase addon all` can generate kube-proxy before `waitForPods`.

---
How to reproduce:
1. minikube start
2. ctrl+c after vm is created
3. minikube start again

### CODE REVIEW

**Comment by liubog2008** [comment]
I signed it

**Comment by neerdoc** [comment]
Looks like a typo in the PR. 
`sudo kubeadm init phase addons all --config %s` should be 
`sudo kubeadm init phase addon all --config %s`,
i.e., **addon** not **addons**.

At least my _kubeadm_ complains on it.

**Comment by liubog2008** [comment]
thanks @neerdoc

**Comment by tstromberg** [comment]
Thanks for the PR! Do you mind elaborating a bit on the "why" of this change in the PR description?

I'm not sure what this does compared to the existing commands, and I'm additionally unsure what the connection between this PR and #3936 is. Thank you!

**Comment by liubog2008** [comment]
@tstromberg That's my first comment.
https://github.com/kubernetes/minikube/issues/3843#issuecomment-473563776

**Comment by liubog2008** [comment]
@tstromberg

File: pkg/minikube/bootstrapper/kubeadm/kubeadm.go [hunk 1]
```diff
@@ -264,11 +264,11 @@ func addAddons(files *[]assets.CopyableFile, data interface{}) error {
 }
 
 // waitForPods waits until the important Kubernetes pods are in running state
-func waitForPods(k8s config.KubernetesConfig, quiet bool) error {
+func waitForPods(k8s config.KubernetesConfig, quiet bool, componentsOnly bool) error {
```
Comment (tstromberg) [inline pkg/minikube/bootstrapper/kubeadm/kubeadm.go]: Having two bool arguments to a function is a bad code smell. Do you mind splitting this out into two functions? Perhaps something like:

waitForComponentPods and waitFork8sPods?

Comment (liubog2008) [reply pkg/minikube/bootstrapper/kubeadm/kubeadm.go]: OK

**Comment by tstromberg** [comment]
Any luck splitting waitForPods into two functions?

**Comment by tstromberg** [comment]
Since there haven't been any updates to this, I've proposed a simpler solution in #4276 

Thank you @liubog2008 for pointing us in the right direction for solving this nefarious problem.

**Pull request closed without merging**